### PR TITLE
Add issue references to Issue model

### PR DESCRIPTION
**Pull request by avkranenburg**
This adds the short, relative and full reference for an issue.

### CODE REVIEW

File: NGitLab/Models/References.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Text.Json.Serialization;
+
+namespace NGitLab.Models;
+
+public class References
+{
+    [JsonPropertyName("short")]
+    public string Short;
```
Comment (louis-z) [inline NGitLab/Models/References.cs]: Please define these public fields as properties instead.

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Models/References.cs (hunk 1 above)
Comment (avkranenburg) [reply NGitLab/Models/References.cs]: Done, but is there a reason why other classes like User, Issue, Author, Assignee and Milestone are not using properties? (I used fields to be consistent)

Comment (louis-z) [reply NGitLab/Models/References.cs]: Older types use fields but when introducing new types we try to enforce use of properties.

File: NGitLab/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -2117,6 +2117,7 @@ NGitLab.Models.Issue.MergeRequestsCount.get -> int
 NGitLab.Models.Issue.MergeRequestsCount.set -> void
 NGitLab.Models.Issue.Milestone -> NGitLab.Models.Milestone
 NGitLab.Models.Issue.MovedToId -> int?
+NGitLab.Models.Issue.References -> NGitLab.Models.References
```
Comment (louis-z) [inline NGitLab/PublicAPI.Unshipped.txt]: @avkranenburg, you're still missing info in this file. Also, this line is not at the correct location; it should be right after the next one, to remain sorted.

If you're using Visual Studio, I would recommend that you
- Undo all changes to the file (i.e. bring it back to the state it's in in the `main` branch)
- Use the IDE's embedded refactoring and select "Fix all occurrences in: **Solution**"

image

If you're using VS Code, Rider or something else, I'm not sure how to achieve the same effect.

Comment (avkranenburg) [reply NGitLab/PublicAPI.Unshipped.txt]: I don't use it often but do have it installed, will use that!

**Review by louis-z — APPROVED** [review]

**Pull request merged**